apps/forklift/lib/forklift/jobs/job_utils.ex

```diff
@@ -0,0 +1,36 @@
+defmodule Forklift.Jobs.JobUtils do
+  @moduledoc false
+  use Retry.Annotation
+  alias Pipeline.Writer.TableWriter.Helper.PrestigeHelper
+
+  @retries Application.get_env(:forklift, :compaction_retries, 10)
+  @backoff Application.get_env(:forklift, :compaction_backoff, 10)
+
+  @retry with: constant_backoff(@backoff) |> Stream.take(@retries)
+  def verify_count(table, target_count, message) do
+    with {:ok, actual_count} <- PrestigeHelper.count(table),
+         {:ok, _} <- check_count(actual_count, target_count) do
+      {:ok, actual_count}
+    else
+      {:error, actual_count} when is_number(actual_count) ->
+        {:error,
+         "Table #{table} with count #{actual_count} did not match expected record count of #{target_count} while trying to verify that #{
+           message
+         }"}
+
+      {:error, error} ->
+        {:error,
+         "Could not verify record count of table #{table} while trying to verify that #{message}: #{inspect(error)}"}
+    end
+  end
+
+  defp check_count(actual_count, target_count) do
```

Comment:
fun facts. not to say you should or shouldn't, but this can also be done with

```elixir
defp check_count(match_count, match_count), do: true
defp check_count(_, _), do: false
```

Reply:
That is a fun but extremely confusing fact. I'll leave it as is for now, this process is complex enough as it is without elixir trickery.